samcli/commands/pipeline/bootstrap/cli.py

```diff
@@ -0,0 +1,221 @@
+"""
+CLI command for "pipeline bootstrap", which sets up the require pipeline infrastructure resources
+"""
+import os
+from typing import Any, cast, Dict, List, Optional
+
+import click
+
+from samcli.cli.cli_config_file import configuration_option, TomlProvider
+from samcli.cli.context import get_cmd_names
+from samcli.cli.main import pass_context, common_options, aws_creds_options, print_cmdline_args
+from samcli.lib.config.samconfig import SamConfig
+from samcli.lib.pipeline.bootstrap.stage import Stage
+from samcli.lib.telemetry.metric import track_command
+from samcli.lib.utils.version_checker import check_newer_version
+from .guided_context import GuidedContext
+
+SHORT_HELP = "Sets up infrastructure resources for AWS SAM CI/CD pipelines."
+
+HELP_TEXT = """Sets up the following infrastructure resources for AWS SAM CI/CD pipelines:
+\n\t - Pipeline user(IAM User) with AccessKeyId and SecretAccessKey credentials to be shared with the CI/CD provider
+\n\t - Pipeline Execution Role(IAM Role) that is assumed by the Pipeline user to obtain access to the AWS account
+\n\t - CloudFormation Execution Role(IAM Role) that is assumed by CloudFormation to deploy the SAM application
+\n\t - Artifacts bucket(S3 bucket) to store the sam build artifacts
+\n\t - ECR repo for the container images of Lambda functions having PackageType property set to Image
+"""
+
+PIPELINE_CONFIG_DIR = os.path.join(".aws-sam", "pipeline")
+PIPELINE_CONFIG_FILENAME = "pipelineconfig.toml"
+
+
+@click.command("bootstrap", short_help=SHORT_HELP, help=HELP_TEXT, context_settings=dict(max_content_width=120))
+@configuration_option(provider=TomlProvider(section="parameters"))
+@click.option(
+    "--interactive/--no-interactive",
+    is_flag=True,
+    default=True,
+    help="Disable interactive prompting for bootstrap parameters, and fail if any required arguments are missing.",
+)
+@click.option(
+    "--stage-name",
+    help="The name of the corresponding pipeline stage. It is used as a suffix for the created resources.",
+    required=False,
+)
+@click.option(
+    "--pipeline-user",
+    help="The ARN of the IAM user having its AccessKeyId and SecretAccessKey shared with the CI/CD provider."
+    "It is used to grant this IAM user the permissions to access the corresponding AWS account."
+    "If not provided, the command will create one along with AccessKeyId and SecretAccessKey credentials.",
+    required=False,
+)
+@click.option(
+    "--pipeline-execution-role",
+    help="The ARN of an IAM Role to be assumed by the pipeline-user to operate on this stage. "
+    "Provide it only if you want to user your own role, otherwise, the command will create one",
+    required=False,
+)
+@click.option(
+    "--cloudformation-execution-role",
+    help="The ARN of an IAM Role to be assumed by the CloudFormation service while deploying the application's stack "
```

Comment:
```suggestion
    help="The ARN of an IAM Role to be assumed by the CloudFormation service while deploying the application's stack. "
```

<img width="932" alt="image" src="https://user-images.githubusercontent.com/3804518/114950153-543a5200-9e07-11eb-956e-aec2f64d2f48.png">